src/plots/plots.js

```diff
@@ -468,7 +477,29 @@ plots.supplyDefaults = function(gd) {
 
     // first fill in what we can of layout without looking at data
     // because fullData needs a few things from layout
-    plots.supplyLayoutGlobalDefaults(newLayout, newFullLayout);
+
+    if(oldFullLayout._initialAutoSizeIsDone) {
+        // coerce the updated layout while preserving width and height
+        var oldWidth = oldFullLayout.width,
```

Comment:
Oh I see that you deleted that block.

I guess one less place where we `gd.layout` is mutated is a good thing. :+1: